src/clj_puppetdb/http.clj

```diff
@@ -24,47 +24,86 @@
   `puppetlabs.ssl-utils.core/pems->ssl-context` function."
   [:ssl-cert :ssl-key :ssl-ca-cert])
 
+(defn- make-client-common
+  [^String host opts]
+  (let [opts (assoc opts :as :stream)
+        info {:host host}
+        info (if-let [ssl-context (:ssl-context opts)]
+               (assoc info :ssl-context ssl-context)
+               info)]
+    (fn
+      ; This arity-overloaded variant of the function is the one called by the `GET` function.
+      ; The `this` parameter refers under normal circumstances to this very function and it is
+      ; needed to be able to call another arity-overloaded variant of the function from the
+      ; function body. When the `this` parameter refers to a different function then it
+      ; effectively overrides the other arity-overloaded variants of the function.
+      ([this path params]
```

Comment:
You can name an anonymous function i.e.

(fn xxyyzz

The recursive call becomes:

(xxyyzz (.toString query))))

Then there is no need to pass "this" in at all, which makes this clearer to my eyes, and also resolves Karel's comment about (client client...) below.